citrination_client/search/client.py

```diff
@@ -30,6 +32,19 @@ def _handle_response(self, response, failure_message=DEFAULT_FAILURE_MESSAGE):
 
         return super(SearchClient, self)._handle_response(response, failure_message)
 
+    def _validate_query_depth(self, pif_system_returning_query):
+        """
+        Checks to see that the query will not exceed the max query depth
+
+        :param pif_system_returning_query: The PIF system query to execute.
+        :type pif_system_returning_query: :class:`PifSystemReturningQuery`
+        """
+
+        start_index = pif_system_returning_query.from_index or 0
+        size = pif_system_returning_query.size or 0
+        if start_index + size > MAX_QUERY_DEPTH:
```

Comment:
You could also a check here to make sure start_index and size are non-negative integers, though maybe they've already been validated somewhere else.